pkg/controller/controlplane/valuesprovider.go

```diff
@@ -503,3 +514,40 @@ func getNetworkNames(
 
 	return networkName, subNetworkName
 }
+
+func (vp *valuesProvider) isOverlayEnabled(network v1beta1.Networking) (bool, error) {
+	if network.ProviderConfig == nil {
+		return true, nil
+	}
+
+	// should not happen in practice because we will receive a RawExtension with Raw populated in production.
+	networkProviderConfig, err := network.ProviderConfig.MarshalJSON()
+	if err != nil {
+		return false, err
+	}
+	if string(networkProviderConfig) == "null" {
+		return true, nil
+	}
+
+	switch network.Type {
+	case calico.ReleaseName:
+		networkConfig := &calicov1alpha1.NetworkConfig{}
+		if _, _, err := vp.Decoder().Decode(networkProviderConfig, nil, networkConfig); err != nil {
+			return false, err
+		}
+		o := networkConfig.Overlay
+		return o == nil || o.Enabled, nil
+	case cilium.ReleaseName:
+		networkConfig := &ciliumv1alpha1.NetworkConfig{}
+		if _, _, err := vp.Decoder().Decode(networkProviderConfig, nil, networkConfig); err != nil {
+			return false, err
+		}
+		o := networkConfig.Overlay
+		if o == nil {
+			return true, nil
+		}
+		return o == nil || o.Enabled, nil
+	}
+
+	return true, nil
```

Comment:
Would it not be a good idea to return an error here? This is certainly not an expected case in that we do not know the networking type.

Reply:
Not sure. That would strictly limit the networking extensions to the very limited set of {calico,cilium} and hardcode that dependency on the infra provider.

Reply:
Sure, your call.